### ISSUE TITLE
Terrain tiles `eval` issues

### ISSUE THREAD

**Issue by dsullivan-creare** [creare-com/podpac#408]
**Description**
TerrainTiles node will not evaluate my coordinates

**Steps to Reproduce**
Try to evaluate a TerrainTiles node for a couple of `Coordinates` objects, as follows:
```python
# instantiate node:
from podpac.datalib.modis_pds import MODIS
terrain = podpac.datalib.terraintiles.TerrainTiles(zoom=6)

# create some coordinates
lat = podpac.crange(  60,  10, -2.0)  # (start, stop, step)
lon = podpac.crange(-130, -60,  2.0)  # (start, stop, step)
time = '2018-05-19T12:00:00'

# Create the PODPAC Coordinates
lat_lon_time = podpac.Coordinates([lat, lon, time], dims=['lat', 'lon', 'time'])
# stacked lat/lon monthly:
lat2 = podpac.clinspace(0, 50, 5)
lon2 = podpac.clinspace(-130, -80, 5)
time2 = podpac.crange('2018-05-19', '2018-06-19', '1,D')

stacked_monthly = podpac.Coordinates([[lat2, lon2], time2], dims=[['lat', 'lon'], 'time'])

# first attempt:
terrain.eval(lat_lon_time)
# results in:
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in _handle_msg(self, msg)
    674                 if 'buffer_paths' in data:
    675                     _put_buffers(state, data['buffer_paths'], msg['buffers'])
--> 676                 self.set_state(state)
    677 
    678         # Handle a state request.

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in set_state(self, sync_data)
    543                     from_json = self.trait_metadata(name, 'from_json',
    544                                                     self._trait_from_json)
--> 545                     self.set_trait(name, from_json(sync_data[name], self))
    546 
    547     def send(self, content, buffers=None):

~/anaconda3/envs/soilmap/lib/python3.6/contextlib.py in __exit__(self, type, value, traceback)
     86         if type is None:
     87             try:
---> 88                 next(self.gen)
     89             except StopIteration:
     90                 return False

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in hold_trait_notifications(self)
   1129                 for changes in cache.values():
   1130                     for change in changes:
-> 1131                         self.notify_change(change)
   1132 
   1133     def _notify_trait(self, name, old_value, new_value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    604                 # Send new state to front-end
    605                 self.send_state(key=name)
--> 606         super(Widget, self).notify_change(change)
    607 
    608     def __repr__(self):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1174                 c = getattr(self, c.name)
   1175 
-> 1176             c(change)
   1177 
   1178     def _add_notifiers(self, handler, name, type):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget_selection.py in _propagate_index(self, change)
    233             self.label = label
    234         if self.value is not value:
--> 235             self.value = value
    236 
    237     @validate('value')

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in __set__(self, obj, value)
    583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
--> 585             self.set(obj, value)
    586 
    587     def _validate(self, obj, value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in set(self, obj, value)
    572             # we explicitly compare silent to True just in case the equality
    573             # comparison above returns something other than True/False
--> 574             obj._notify_trait(self.name, old_value, new_value)
    575 
    576     def __set__(self, obj, value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in _notify_trait(self, name, old_value, new_value)
   1137             new=new_value,
   1138             owner=self,
-> 1139             type='change',
   1140         ))
   1141 

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    604                 # Send new state to front-end
    605                 self.send_state(key=name)
--> 606         super(Widget, self).notify_change(change)
    607 
    608     def __repr__(self):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1174                 c = getattr(self, c.name)
   1175 
-> 1176             c(change)
   1177 
   1178     def _add_notifiers(self, handler, name, type):

<ipython-input-11-ade2458d562c> in coords_on_change(change)
     50 def coords_on_change(change):
     51     if change['type'] == 'change' and change['name'] == 'value':
---> 52         o = test_nodes[layers_dropdown.value - 1].eval(test_coordinates[change['new'] - 1])
     53 
     54 #         eval_coords = test_coords[layers_dropdown.label][change['new'] - 1]

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/node.py in wrapper(self, coordinates, output)
    982             self._from_cache = True
    983         else:
--> 984             data = fn(self, coordinates, output=output)
    985             if self.cache_output:
    986                 self.put_cache(data, key, cache_coordinates)

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/compositor/compositor.py in eval(self, coordinates, output)
    239         self._requested_coordinates = coordinates
    240         outputs = self.iteroutputs(coordinates)
--> 241         output = self.composite(coordinates, outputs, output)
    242         return output
    243 

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/compositor/ordered_compositor.py in composite(self, coordinates, data_arrays, result)
     51         for data in data_arrays:
     52             if self.outputs is None:
---> 53                 data = data.transpose(*result.dims)
     54                 self._composite(result, data, mask)
     55             else:

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/xarray/core/dataarray.py in transpose(self, transpose_coords, *dims)
   1900         """
   1901         if dims:
-> 1902             dims = tuple(utils.infix_dims(dims, self.dims))
   1903         variable = self.variable.transpose(*dims)
   1904         if transpose_coords:

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/xarray/core/utils.py in infix_dims(dims_supplied, dims_all)
    687         if set(dims_supplied) ^ set(dims_all):
    688             raise ValueError(
--> 689                 f"{dims_supplied} must be a permuted list of {dims_all}, unless `...` is included"
    690             )
    691         yield from dims_supplied

ValueError: ('lat', 'lon', 'time') must be a permuted list of ('lat', 'lon'), unless `...` is included
```

The stacked coordinates also fail:

```python
# second attempt:
terrain.eval(stacked_monthly)

# results in:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in _handle_msg(self, msg)
    674                 if 'buffer_paths' in data:
    675                     _put_buffers(state, data['buffer_paths'], msg['buffers'])
--> 676                 self.set_state(state)
    677 
    678         # Handle a state request.

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in set_state(self, sync_data)
    543                     from_json = self.trait_metadata(name, 'from_json',
    544                                                     self._trait_from_json)
--> 545                     self.set_trait(name, from_json(sync_data[name], self))
    546 
    547     def send(self, content, buffers=None):

~/anaconda3/envs/soilmap/lib/python3.6/contextlib.py in __exit__(self, type, value, traceback)
     86         if type is None:
     87             try:
---> 88                 next(self.gen)
     89             except StopIteration:
     90                 return False

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in hold_trait_notifications(self)
   1129                 for changes in cache.values():
   1130                     for change in changes:
-> 1131                         self.notify_change(change)
   1132 
   1133     def _notify_trait(self, name, old_value, new_value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    604                 # Send new state to front-end
    605                 self.send_state(key=name)
--> 606         super(Widget, self).notify_change(change)
    607 
    608     def __repr__(self):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1174                 c = getattr(self, c.name)
   1175 
-> 1176             c(change)
   1177 
   1178     def _add_notifiers(self, handler, name, type):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget_selection.py in _propagate_index(self, change)
    233             self.label = label
    234         if self.value is not value:
--> 235             self.value = value
    236 
    237     @validate('value')

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in __set__(self, obj, value)
    583             raise TraitError('The "%s" trait is read-only.' % self.name)
    584         else:
--> 585             self.set(obj, value)
    586 
    587     def _validate(self, obj, value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in set(self, obj, value)
    572             # we explicitly compare silent to True just in case the equality
    573             # comparison above returns something other than True/False
--> 574             obj._notify_trait(self.name, old_value, new_value)
    575 
    576     def __set__(self, obj, value):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in _notify_trait(self, name, old_value, new_value)
   1137             new=new_value,
   1138             owner=self,
-> 1139             type='change',
   1140         ))
   1141 

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/ipywidgets/widgets/widget.py in notify_change(self, change)
    604                 # Send new state to front-end
    605                 self.send_state(key=name)
--> 606         super(Widget, self).notify_change(change)
    607 
    608     def __repr__(self):

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/traitlets/traitlets.py in notify_change(self, change)
   1174                 c = getattr(self, c.name)
   1175 
-> 1176             c(change)
   1177 
   1178     def _add_notifiers(self, handler, name, type):

<ipython-input-12-ade2458d562c> in coords_on_change(change)
     50 def coords_on_change(change):
     51     if change['type'] == 'change' and change['name'] == 'value':
---> 52         o = test_nodes[layers_dropdown.value - 1].eval(test_coordinates[change['new'] - 1])
     53 
     54 #         eval_coords = test_coords[layers_dropdown.label][change['new'] - 1]

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/node.py in wrapper(self, coordinates, output)
    982             self._from_cache = True
    983         else:
--> 984             data = fn(self, coordinates, output=output)
    985             if self.cache_output:
    986                 self.put_cache(data, key, cache_coordinates)

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/compositor/compositor.py in eval(self, coordinates, output)
    239         self._requested_coordinates = coordinates
    240         outputs = self.iteroutputs(coordinates)
--> 241         output = self.composite(coordinates, outputs, output)
    242         return output
    243 

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/compositor/ordered_compositor.py in composite(self, coordinates, data_arrays, result)
     49 
     50         mask = UnitsDataArray.create(coordinates, outputs=self.outputs, data=0, dtype=bool)
---> 51         for data in data_arrays:
     52             if self.outputs is None:
     53                 data = data.transpose(*result.dims)

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/core/compositor/compositor.py in iteroutputs(self, coordinates)
    188 
    189         # get sources, potentially downselected
--> 190         sources = self.select_sources(coordinates)
    191 
    192         if settings["DEBUG"]:

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/datalib/terraintiles.py in select_sources(self, coordinates)
    147     def select_sources(self, coordinates):
    148         # get all the tile sources for the requested zoom level and coordinates
--> 149         sources = get_tile_urls(self.tile_format, self.zoom, coordinates)
    150 
    151         # create TerrainTilesSource classes for each url source

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/datalib/terraintiles.py in get_tile_urls(tile_format, zoom, coordinates)
    196 
    197     # get all the tile definitions for the requested zoom level
--> 198     tiles = _get_tile_tuples(zoom, coordinates)
    199 
    200     # get source urls

~/anaconda3/envs/soilmap/lib/python3.6/site-packages/podpac/datalib/terraintiles.py in _get_tile_tuples(zoom, coordinates)
    241 
    242         if "lat" not in coordinates or "lon" not in coordinates:
--> 243             raise TypeError("input coordinates must have lat and lon dimensions to get tiles")
    244 
    245         # transform to WGS84 (epsg:4326) to use the mapzen example for transforming coordinates to tilespace

TypeError: input coordinates must have lat and lon dimensions to get tiles

```

**Comment by dsullivan-creare** [comment]
@jmilloy Can you confirm whether I'm doing some incorrectly, or this is a bug?

**Comment by jmilloy** [comment]
Yes, this is fixed but was not yet merged into develop. See #406 and #407. I just merged #407 so your `lat_lon_time` coordinates should now work.
